app/scripts/metamask-controller.test.js

```diff
@@ -976,36 +1094,802 @@ describe('MetaMaskController', () => {
+                origin: 'npm:snap',
+              },
+              permissions: {
+                [PermissionNames.eth_accounts]: {
+                  caveats: [
+                    {
+                      type: CaveatTypes.restrictReturnedAccounts,
+                      value: ['foo'],
+                    },
+                  ],
+                },
+              },
+            },
+            type: 'wallet_requestPermissions',
+          }),
+        );
+      });
+
+      it('throws an error if the eth_accounts and permittedChains approval is rejected', async () => {
+        jest
+          .spyOn(
+            metamaskController.approvalController,
+            'addAndShowApprovalRequest',
+          )
+          .mockRejectedValue(new Error('approval rejected'));
+
+        await expect(() =>
+          metamaskController.requestCaip25Permission('test.com', {
+            eth_accounts: {},
+          }),
+        ).rejects.toThrow(new Error('approval rejected'));
+      });
+
+      it('grants the CAIP-25 permission with eth accounts, chainIds, and isMultichainOrigin: false if origin is not snapId', async () => {
+        jest
+          .spyOn(
+            metamaskController.approvalController,
+            'addAndShowApprovalRequest',
+          )
+          .mockResolvedValue({
+            approvedChainIds: ['0x1', '0x5'],
+            approvedAccounts: ['0xdeadbeef'],
+          });
+        jest
+          .spyOn(metamaskController.permissionController, 'grantPermissions')
+          .mockReturnValue({
+            [Caip25EndowmentPermissionName]: {
+              foo: 'bar',
+            },
+          });
+
+        await metamaskController.requestCaip25Permission('test.com', {});
+
+        expect(
+          metamaskController.permissionController.grantPermissions,
+        ).toHaveBeenCalledWith({
+          subject: { origin: 'test.com' },
+          approvedPermissions: {
+            [Caip25EndowmentPermissionName]: {
+              caveats: [
+                {
+                  type: Caip25CaveatType,
+                  value: {
+                    requiredScopes: {},
+                    optionalScopes: {
+                      'eip155:1': {
+                        accounts: ['eip155:1:0xdeadbeef'],
+                      },
+                      'eip155:5': {
+                        accounts: ['eip155:5:0xdeadbeef'],
+                      },
+                    },
+                    isMultichainOrigin: false,
+                  },
+                },
+              ],
+            },
+          },
+        });
+      });
+
+      it('grants the CAIP-25 permission approved accounts for the `wallet:eip155` scope with isMultichainOrigin: false if origin is snapId', async () => {
+        jest
+          .spyOn(
+            metamaskController.approvalController,
+            'addAndShowApprovalRequest',
+          )
+          .mockResolvedValue({
+            approvedChainIds: ['0x1', '0x5'],
```

Comment:
```suggestion
            approvedChainIds: ['0x1', '0x5'],
```

presumably this isn't currently possible?

Reply:
yeah, the UI should not showing an Approval veiw that can return approvedChainIds, but I also explicitly filter out / ignore them in code. Is it worth leaving this in to be doubly sure or do you think it just causes confusion?

Reply:
No its okay!